morpho5/utils/common.h

```diff
@@ -25,10 +25,23 @@
  * Functions and macros for comparing values
  * ----------------------------------------- */
 
+/** @brief Compares two values
+ * @param l value to compare
+ * @param r value to compare */
+#define MORPHO_CHECKCMPTYPE(l, r) \
```

Comment:
Perhaps a better name for this would be MORPHO_CASTNUMERICALTYPE as this isn't really checking if the types are comparable and returning a bool, but casting integers to float for comparison

Reply:
I changed it to MORPHO_CMPPROMOTETYPE